include/swift/Option/Options.td

```diff
@@ -443,6 +443,9 @@ def emit_parseable_module_interface_path :
 def emit_objc_header : Flag<["-"], "emit-objc-header">,
   Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>,
   HelpText<"Emit an Objective-C header file">;
+def objc_header_is_public : Flag<["-"], "objc-header-is-public">,
+  Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>,
+  HelpText<"Objective-C header file will be used as public APIs">;
```

Comment:
Suggested help text:
```
  HelpText<"Omit internal declarations from the Objective-C header file">;
```

I'm not 100% happy with the flag name either, but I'm not sure what would be better.